dz2.1/ДЗ. TypeScript. Анотація типів та примітивні типи.ts

```diff
@@ -0,0 +1,116 @@
+class School {
+  directions: Direction[];
+
+  addDirection(direction: Direction) {
+    this.directions.push(direction);
+  }
+}
+
+class Direction {
+  levels: Level[];
+
+  get name() {
+    return this._name;
+  }
+
+  constructor(name: string) {
+    this._name = name;
+  }
+
+  addLevel(level: Level) {
+    this.levels.push(level);
+  }
+}
+
+class Level {
+  groups: Group[];
+
+  constructor(name: string, program: string) {
+    this.name = name;
+    this._program = program;
+  }
+
+  get name(): string {
+    return this._name;
+  }
+
+  get program(): string {
+    return this._program;
+  }
+
+  addGroup(group: Group) {
+    this.groups.push(group);
+  }
+}
+
+class Group {
+  _students: Student[];
+
+  get students(): Student[] {
+    return this._students;
+  }
+
+  constructor(directionName: string, levelName: string) {
+    this.directionName = directionName;
+    this.levelName = levelName;
+  }
+
+  addStudent(student: Student) {
+    this._students.push(student);
+  }
+
+  showPerformance(): Student[] {
+    const sortedStudents = this.students.toSorted(
+      (a: number, b: number) => b.getPerformanceRating() - a.getPerformanceRating()
+    );
+
+    return sortedStudents;
+  }
+}
+
+class Student {
+  grades: number[];
+  attendance: boolean[];
+
+  constructor(firstName: string, lastName: string, birthYear: number) {
+    this.firstName = firstName;
+    this.lastName = lastName;
+    this.birthYear = birthYear;
+  }
+
+  get fullName(): string {
+    return `${this.lastName} ${this.firstName}`;
+  }
+
+  set fullName(value: string) {
+    [this.lastName, this.firstName] = value.split(" ");
+  }
+
+  get age(): number {
+    return new Date().getFullYear() - this.birthYear;
+  }
+
+  setGrade(subject: string, grade: string) {
```

Comment:
Please, do not forget to explicitly add type annotation for the return function value.